src/app/storage/duck/sagas.ts

```diff
@@ -0,0 +1,265 @@
+import { takeLatest, select, race, take, call, put, delay, } from 'redux-saga/effects';
+import { ClientFactory } from '../../../client/client_factory';
+import { IClusterClient } from '../../../client/client';
+import { MigResource, MigResourceKind } from '../../../client/resources';
+import {
+  CoreNamespacedResource,
+  CoreNamespacedResourceKind,
+} from '../../../client/resources';
+import {
+  createStorageSecret,
+  createMigStorage,
+  updateMigStorage,
+  updateStorageSecret,
+} from '../../../client/resources/conversions';
+import { Creators } from './actions';
+import { alertErrorTimeout } from '../../common/duck/actions';
+import {
+  createAddEditStatus,
+  AddEditState,
+  AddEditMode,
+  IAddEditStatus,
+  AddEditWatchTimeout,
+  AddEditWatchTimeoutPollInterval,
+  AddEditConditionCritical,
+  createAddEditStatusWithMeta,
+  AddEditConditionReady,
+  AddEditDebounceWait,
+} from '../../common/add_edit_state';
+
+function* addStorageRequest(action)  {
+  // TODO: Need to improve this to fall into the failed create state with rollback
```

Comment:
There is sometimes a bit of a delay when clicking the add button & seeing the button disabled. I think you could add some ui state to show isLoading as soon as the add/update is dispatched for storage/cluster. This could be a candidate for a later issue/refactor.